pkg/providers/oidc/provider_verifier.go

```diff
@@ -79,6 +79,19 @@ func (p ProviderVerifierOptions) toOIDCConfig() *oidc.Config {
 		ClientID:          p.ClientID,
 		SkipIssuerCheck:   p.SkipIssuerVerification,
 		SkipClientIDCheck: true,
+		// support all signing algorithms for backwards compatibility
+		// there is no security value in this config, either you trust the issuer or you do not
```

Comment:
Why do we not want to limit this to explicitly those supported by the provider? Surely given we have this information from the discovery document we should pass it through here?